integrationtests/src/test/java/org/axonframework/integrationtests/eventhandling/TrackingEventProcessorTest.java

```diff
@@ -1849,6 +1849,50 @@ public void close() {
         assertWithin(20, TimeUnit.MILLISECONDS, () -> assertTrue(testSubject.processingStatus().get(0).isCaughtUp()));
     }
 
+    @Test
+    @Timeout(value = 1000)
+    void testIsReplayingWhenNotCaughtUp() throws Exception {
+        when(mockHandler.supportsReset()).thenReturn(true);
+        final List<String> handled = new CopyOnWriteArrayList<>();
```

Comment:
The `handled`, `handledInRedelivery` and `doAnswer` sections seem to not be used for this test case.

Guessing that just publishing the events, initiating the replay and testing the `isReplaying` outcome is sufficient.

Reply:
It does not work without these properties